src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -1055,6 +1055,89 @@ object Desugar {
               }
             } yield in.Seqn(Vector(dPre, exprAss, clauseBody))(src)
 
+          case n: POutline => 
+            val src: Meta = meta(n, info)
+            val variables = info.variables(n.body).toVector
+            val modified = info.modified(n.body).toVector
+            val declared = info.declared(n.body).toVector
+            // create in parameters
+            val allArgs = variables.map {id => 
+              in.Parameter.In(idName(id, info), typeD(info.typ(id), Addressability.inParameter)(src))(src)
+            }
+            val modifiedArgsWithSubsAux = modified.map {id => 
+              val param = in.Parameter.In(idName(id, info), typeD(info.typ(id), Addressability.inParameter)(src))(src)
+              val local = localAlias(localVarContextFreeD(id, info))
+              (param, local)
+            }
+            val (_, modifiedArgsSubs) = modifiedArgsWithSubsAux.unzip
+            // create out parameters
+            val returns = (modified ++ declared).zipWithIndex.map { case (id, idx) => 
+              in.Parameter.Out(nm.outParam(idx, info.codeRoot(n), info), typeD(info.typ(id), Addressability.outParameter)(src))(src)
+            }
+            // add temporary return variables to initialization
+            val targets = (modified ++ declared).map(id => 
+              freshExclusiveVar(typeD(info.typ(id), Addressability.callResult)(src))(src)
+            )
+            // add declared and modified variables to initalization
+            val declaredLocalVars = declared.map(id => 
+              in.LocalVar(idName(id, info), typeD(info.typ(id), Addressability.outParameter)(src))(src)
+            )
+            val modifiedLocalVars = modified.map(id => 
+              in.LocalVar(idName(id, info), typeD(info.typ(id), Addressability.outParameter)(src))(src)
+            )
+            val blockInit = targets ++ declaredLocalVars
+            // create arguments for function call
+            val outArgs = variables.map(id => in.LocalVar(idName(id, info),typeD(info.typ(id), Addressability.outParameter)(src))(src))
+            // add function call
+            val proxy = in.FunctionProxy(nm.function("outline"++nm.fresh, info))(meta(n, info))
```

Comment:
You can even add a new function to the `NameManager` that generates names for outline blocks